### PR TITLE
Debug frame Metal Documentation + doc links

### PR DESCRIPTION
**Pull request by CedricGuillemet**

### CODE REVIEW

**Comment by bghgary** [comment]
@ryantrem @Alex-MSFT Can one of you try to follow the instructions and see if it works for you?

File: Documentation/DebugRenderedFrameMetal.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+![CaptureResult](Images/FrameCaptureiOS/CaptureResult.jpg)
+
+On the left: draw calls and resources binding (vertex buffers, textures, render pipelines,...)
+Top middle: vertex and fragment pipelines view for the draw call
+Bottom middle: list of resources for the draw call
+Top right: result of the draw call for all the bound render targets
+Bottom right: debug output
+
+As RenderDoc, you debug by inspectin draw calls; where resources are commited to rasterize triangles.
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:20]: nit: inspecting

Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:20]: As with RenderDoc?

File: Documentation/DebugRenderedFrameMetal.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Inputs streams (positions, UV, normals) are transformed by the vertex shader to produce (usualy) screen space streams.
+Those streams are used to render triangles where each pixel passes trough a fragment shader that returns a color.
+Vertex and fragment shaders processing are parametrized by uniforms (matrices that convert from world space to screen space, light position and orientation, ...)
+
+![CaptureDrawCall](Images/FrameCaptureiOS/CaptureDrawCall.jpg)
+
+When a rendering goes wrong, one or multiple parts of that pipeline are also wrong.
+If nothing is displayed, check you have all the resources (shaders, vertex buffer, textures,...) are bound.
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:28]: Nit: remove 'are' or remove 'you have'

File: Documentation/DebugRenderedFrameMetal.md (hunk 1 above)
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:20]:
```suggestion
As with RenderDoc, debug by inspecting draw calls; where resources are commited to rasterize triangles.
```

File: Documentation/DebugRenderedFrameMetal.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+First thing to check is the geometry. It's common to expect a mesh to be rendered but you render something else because of previous stream is not replaced for example.
+If you double click the geometry in the vertex view, you'll see the mesh after transformation by the vertex shader that is used for the draw call:
+
+![CaptureGeometry](Images/FrameCaptureiOS/CaptureGeometry.jpg)
+
+The white rectangle corresponds to the display view.
+See the vertex stream dumped in the bottom right list.
+If your geometry is completely off screen you should see it here.
+
+You can modify the virtual debug camera and select a particular triangle to see the stream vertices that is producing it.
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:40]:
```suggestion
You can modify the virtual debug camera and select a particular triangle to see the stream vertices that are producing it.
```

File: Documentation/DebugRenderedFrameMetal.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+![CaptureGeometryReview](Images/FrameCaptureiOS/CaptureGeometryReview.jpg)
+
+Also, note the white truncated pyramid that corresponds to the perspective and view matrix.
+
+To check the uniforms, double click on the Buffer 0 both for vertex and fragment shaders.
+The data in that buffer then looks like this:
+
+![CaptureVertexUniforms](Images/FrameCaptureiOS/CaptureVertexUniforms.jpg)
+
+The order of bound values can be check by double clicking on the Vertex Function. The shader source uses a struct ```Frame``` that corresponds to the bound uniform buffer.
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:51]:
```suggestion
The order of bound values can be checked by double clicking on the Vertex Function. The shader source uses a struct ```Frame``` that corresponds to the bound uniform buffer.
```

File: Documentation/DebugRenderedFrameMetal.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+![CaptureVertexShaderSource](Images/FrameCaptureiOS/CaptureVertexShaderSource.jpg)
+
+And it works the same for fragment shader:
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:55]:
```suggestion
And it works the same for fragment shaders:
```

File: Documentation/Readme.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+[Put break points in your JS. Inspect variable and watch the console output on your Mac for iOS and MacOS.](DebugJavascriptMacIOS.md)
+
+## Debugging iOS and MacOS rendering using Xcode
+
+[Inspect and debug your rendering thanks to Xcode frame capture.](DebugRenderedFrameMetal.md)
+
+## Extending Babylon Native
+
+[Babylon Native extensive plugin system allows you to add and expose functionnalities. Here is how.](Extending.md)
```
Comment (Alex-MSFT) [inline Documentation/Readme.md:21]:
```suggestion
[Babylon Native's extensive plugin system allows you to add and expose functionalities. Here is how.](Extending.md)
```

File: Documentation/Readme.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+[Some tips to improve your experience of Babylon  Native with the Android Simulator.](AndroidSimulator.md)
+
+## Build system overview
+
+[Everything you need to know about the build system and the dependencies management.](BuildSystem.md)
+
+## Debugging JavaScript on Mac and iOS
+
+[Put break points in your JS. Inspect variable and watch the console output on your Mac for iOS and MacOS.](DebugJavascriptMacIOS.md)
```
Comment (Alex-MSFT) [inline Documentation/Readme.md:13]:
```suggestion
[Put break points in your JS. Inspect variables and watch the console output on your Mac for iOS and MacOS.](DebugJavascriptMacIOS.md)
```

**Comment by Alex-MSFT** [comment]
> @ryantrem @Alex-MSFT Can one of you try to follow the instructions and see if it works for you?

I took a look, and did some debugging this is a really helpful doc!  I added some suggestions to clean up the wording.

File: Documentation/DebugRenderedFrameMetal.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+On the left: draw calls and resources binding (vertex buffers, textures, render pipelines,...)
+Top middle: vertex and fragment pipelines view for the draw call
+Bottom middle: list of resources for the draw call
+Top right: result of the draw call for all the bound render targets
+Bottom right: debug output
+
+As RenderDoc, you debug by inspectin draw calls; where resources are commited to rasterize triangles.
+Inputs streams (positions, UV, normals) are transformed by the vertex shader to produce (usualy) screen space streams.
+Those streams are used to render triangles where each pixel passes trough a fragment shader that returns a color.
```
Comment (Alex-MSFT) [inline Documentation/DebugRenderedFrameMetal.md:22]:
```suggestion
Those streams are used to render triangles where each pixel passes through a fragment shader that returns a color.
```

**Comment by CedricGuillemet** [comment]
Thank you so much @Alex-MSFT for the clean up!

**Pull request merged**